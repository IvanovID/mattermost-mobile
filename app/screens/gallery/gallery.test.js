// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

// import RNFetchBlob from 'rn-fetch-blob';

import Preferences from '@mm-redux/constants/preferences';

import * as NavigationActions from 'app/actions/navigation';

// import BottomSheet from 'app/utils/bottom_sheet';

import ImagePreview from './gallery';

jest.useFakeTimers();
jest.mock('react-intl');
jest.mock('react-native-file-viewer', () => ({
    open: jest.fn(),
}));

describe('ImagePreview', () => {
    const baseProps = {
        canDownloadFiles: true,
        deviceHeight: 400,
        deviceWidth: 300,
        files: [
            {caption: 'Caption 1', source: 'source', data: 'data'},
            {caption: 'Caption 2', source: 'source', data: 'data'},
        ],
        index: 0,
        theme: Preferences.THEMES.default,
        componentId: 'component-id',
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ImagePreview {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, renderDownloadButton', () => {
        const wrapper = shallow(
            <ImagePreview {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.instance().renderDownloadButton()).toMatchSnapshot();
    });

    test('should match state on handleChangeImage', () => {
        const wrapper = shallow(
            <ImagePreview {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.setState({index: 0});
        wrapper.instance().handlePageSelected(1);

        expect(wrapper.state('index')).toEqual(1);
    });

    test('should match call popTopScreen & mergeNavigationOptions on close', () => {
        const mergeNavigationOptions = jest.spyOn(NavigationActions, 'mergeNavigationOptions');
        const popTopScreen = jest.spyOn(NavigationActions, 'popTopScreen');

        const wrapper = shallow(
            <ImagePreview
                {...baseProps}
            />,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.instance().close();

        expect(mergeNavigationOptions).toHaveBeenCalledTimes(1);
        expect(popTopScreen).toHaveBeenCalled();
        expect(mergeNavigationOptions).toHaveBeenCalledWith(
            baseProps.componentId,
            {
                topBar: {
                    visible: true,
                },
            },
        );
    });
});
