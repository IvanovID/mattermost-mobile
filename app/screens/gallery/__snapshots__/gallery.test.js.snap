// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePreview should match snapshot 1`] = `
<React.Fragment>
  <Gallery
    errorComponent={[Function]}
    flatListProps={
      Object {
        "windowSize": 3,
      }
    }
    imageComponent={[Function]}
    images={
      Array [
        Object {
          "caption": "Caption 1",
          "data": "data",
          "source": "source",
        },
        Object {
          "caption": "Caption 2",
          "data": "data",
          "source": "source",
        },
      ]
    }
    initialPage={0}
    onPageSelected={[Function]}
    onSingleTapConfirmed={[Function]}
    onSwipedVertical={[Function]}
    pageMargin={2}
    removeClippedSubviews={true}
    scrollViewStyle={Object {}}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "bottom": 0,
          "opacity": 1,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "#000",
          "height": 99,
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "color": "white",
          "fontSize": 14,
          "marginBottom": 10,
        }
      }
    >
      
    </Text>
  </ForwardRef(AnimatedComponentWrapper)>
  <Downloader
    deviceHeight={400}
    deviceWidth={300}
    file={
      Object {
        "caption": "Caption 1",
        "data": "data",
        "source": "source",
      }
    }
    force={false}
    onCancelPress={[Function]}
    onDownloadCancel={[Function]}
    onDownloadStart={[Function]}
    onDownloadSuccess={[Function]}
    prompt={false}
    saveToCameraRoll={true}
    show={false}
  />
</React.Fragment>
`;

exports[`ImagePreview should match snapshot, renderDownloadButton 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 44,
      "justifyContent": "center",
      "width": 48,
    }
  }
>
  <Icon
    allowFontScaling={false}
    color="#fff"
    name="download-outline"
    size={26}
  />
</ForwardRef>
`;
